### PR TITLE
Unbounded binary search in python

### PR DESCRIPTION
**Pull request by walaa5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(yes, I read them)

### Description

(This is code for unbounded binary search in python .)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#5972)

### CODE REVIEW

File: Python/search/unbounded_binary_search.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""
+Given a function ‘int f(unsigned int x)’ which takes a non-negative integer ‘x’
+as input and returns an integer as output. The function is monotonically increasing
+with respect to value of x, i.e., the value of f(x+1) is greater than f(x) for
+every input x. Find the value ‘n’ where f() becomes positive for the first time.
+Since f() is monotonically increasing, values of f(n+1), f(n+2),…
+must be positive and values of f(n-2), f(n-3), .. must be negative.
+
+Time Complexity : O(log n)
```
Comment (ankitaggarwal23) [inline Python/search/unbounded_binary_search.py]: Shift this at the end of file

File: Python/search/unbounded_binary_search.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+"""
+
+# an example of a function
+# fun(x) = x^2 + 5x - 10
+
```
Comment (ankitaggarwal23) [inline Python/search/unbounded_binary_search.py]: Remove the redundant comments

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Use dynamic inputs from the user, don't use the static one
- Add input-output prompts to make the program more interactive.
-  Add Time and Space Complexity at the end of the Code   
- Add a new line at the EOF

**Comment by walaa5** [comment]
@ankitaggarwal23 , I made the changes, please review them.

File: Python/search/unbounded_binary_search.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      "positive first is : ", out)
+
+
+"""
+Given a function ‘int f(unsigned int x)’ which takes a non-negative integer ‘x’
+as input and returns an integer as output. The function is monotonically increasing
+with respect to value of x, i.e., the value of f(x+1) is greater than f(x) for
+every input x. Find the value ‘n’ where f() becomes positive for the first time.
+Since f() is monotonically increasing, values of f(n+1), f(n+2),…
+must be positive and values of f(n-2), f(n-3), .. must be negative.
```
Comment (ankitaggarwal23) [inline Python/search/unbounded_binary_search.py]: Kindly shift only this at the start

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Write the outputs in the way your code is producing.

File: Python/search/unbounded_binary_search.py (hunk 3 above)
Comment (walaa5) [reply Python/search/unbounded_binary_search.py]: @ankitaggarwal23 ok, I made them , can you see it again.

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by ankitaggarwal23** [comment]
Thank you for your valuable Contribution.😊

**Pull request merged**